arch/arm/src/stm32h7/stm32_serial.c

```diff
@@ -2324,10 +3619,10 @@ void arm_serialinit(void)
   int ret;
 #endif
 
+#ifdef CONFIG_PM
   /* Register to receive power management callbacks */
 
-#ifdef CONFIG_PM
-  ret = pm_register(&g_serialcb);
+  ret = pm_register(&g_serialpm.pm_cb);
```

Comment:
g_serialpm is not defined in this file. Probably `static  struct pm_callback_s g_serialcb` should be replaced with `static struct pm_config_s g_serialpm`

Reply:
@raiden00pl - thank you for the thorough review! I will squash the formatting changes and fix the PM or removed it.

Reply:
@raiden00pl - I pulled in the PM from the F7 which is closer to one that works and have marked it untested.

Reply:
It is squashed ...
```
stm32h7:Serial Apply formatting suggestions from code review

Co-authored-by: Mateusz Szafoni <raiden00pl@gmail.com>

stm32h7: Serail Add Power Managment (Untested)
```